SilKit/source/services/logging/Logger.cpp

```diff
@@ -118,32 +313,124 @@ Logger::Logger(const std::string& participantName, Config::Logging config)
+            else if(sink.type != Config::Sink::Type::Remote)
+            {
+                stdoutSink->set_level(log_level);
+                _loggerSimple->sinks().emplace_back(std::move(stdoutSink));
+            }
+
             break;
         }
         case Config::Sink::Type::File:
         {
-            //
             auto filename = fmt::format("{}_{:%FT%H-%M-%S}.txt", sink.logName, tmBuffer);
             auto fileSink = std::make_shared<spdlog::sinks::basic_file_sink_mt>(filename);
-            fileSink->set_level(log_level);
-            _logger->sinks().push_back(fileSink);
+
+            if (sink.format == Config::Sink::Format::Json)
+            {   
+                using spdlog::details::make_unique; // for pre c++14
+                auto formatter = make_unique<spdlog::pattern_formatter>();
+                formatter->set_pattern(jsonpattern);
+                fileSink->set_pattern(jsonpattern);
+                fileSink->set_level(log_level);
+                _loggerJson->sinks().push_back(fileSink);
+            }
+            else
+            {
+                fileSink->set_level(log_level);
+                _loggerSimple->sinks().push_back(fileSink);
+            }
         }
         }
     }
+    if (nullptr != _loggerSimple)
+    {
+        _loggerSimple->flush_on(to_spdlog(_config.flushLevel));
+    }
+    if (nullptr != _loggerJson)
+    {
+        _loggerJson->flush_on(to_spdlog(_config.flushLevel));
+    }
+}
+
+void Logger::Log(const LoggerMessage& msg)
+{
+    const auto now = log_clock::now();
+    if (nullptr != _loggerJson)
+    {
+        JsonLogMessage myJsonMsg(msg);
+        _loggerJson->log(now, spdlog::source_loc{}, to_spdlog(msg.GetLevel()), fmt::format("{}", myJsonMsg));
+    }
+
+    if (nullptr != _loggerSimple)
+    {
+        SimpleLogMessage myMsg(msg);
+        _loggerSimple->log(now, spdlog::source_loc{}, to_spdlog(msg.GetLevel()), fmt::format("{}", myMsg));
+    }
+    if (nullptr != _loggerRemote)
+    {
+        _loggerRemote->Log(now, msg);
+    }
+}
 
-    _logger->flush_on(to_spdlog(_config.flushLevel));
+void Logger::Log(const LogMsg& msg)
+{
+    LoggerMessage loggerMsg{this, msg};
+    if (nullptr != _loggerJson)
+    {
+            
+        JsonLogMessage jsonMsg{loggerMsg};
+        _loggerJson->log(msg.time, spdlog::source_loc{}, to_spdlog(jsonMsg.m.GetLevel()),
+                            fmt::format("{}", jsonMsg));
+    }
+
+    if (nullptr != _loggerSimple)
+    {
+        SimpleLogMessage simpleMsg{loggerMsg};
+        _loggerSimple->log(msg.time, spdlog::source_loc{}, to_spdlog(simpleMsg.m.GetLevel()),
+                            fmt::format("{}", simpleMsg));
+    }
+    if (nullptr != _loggerRemote)
+    {
+        _loggerRemote->Log(msg);
+    }
 }
 
+
 void Logger::Log(Level level, const std::string& msg)
 {
-    _logger->log(to_spdlog(level), msg);
+    const auto now = log_clock::now();
+    if (nullptr != _loggerJson)
+    {
+        JsonString myJsonString(msg);
```

Comment:
Aggregate initialization ftw!

```suggestion
        JsonString myJsonString{msg};
```